subprojects/core/src/main/java/org/gradle/api/internal/changedetection/changes/DefaultTaskArtifactStateRepository.java

```diff
@@ -153,15 +153,9 @@ public TaskExecutionHistory getExecutionHistory() {
             return this;
         }
 
-        @Nullable
         @Override
-        public UniqueId getOriginBuildInvocationId() {
-            HistoricalTaskExecution previousExecution = history.getPreviousExecution();
-            if (previousExecution == null) {
-                return null;
-            } else {
-                return previousExecution.getBuildInvocationId();
-            }
+        public OriginTaskExecutionMetadata getOriginExecutionMetadata() {
+            return history.getPreviousExecution().getOriginExecutionMetadata();
```

Comment:
`getPreviousExecution()` can never return null here anymore?

Reply:
(Or we never call this in contexts where it returns null?)

Reply:
It can. Fixed it now.